internal/types/subtypes/interceptor_ext_test.go

```diff
@@ -0,0 +1,333 @@
+				Attrs: &attribute.TestNoItemAttributes_Attributes{
+					Attributes: func() *structpb.Struct {
+						attrs, _ := structpb.NewStruct(map[string]interface{}{
+							"name": "test",
+						})
+						return attrs
+					}(),
+				},
+			},
+			&attribute.TestNoItemAttributes{
+				Id: "trsr_one",
+				Attrs: &attribute.TestNoItemAttributes_Attributes{
+					Attributes: func() *structpb.Struct {
+						attrs, _ := structpb.NewStruct(map[string]interface{}{
+							"name": "test",
+						})
+						return attrs
+					}(),
+				},
+			},
+			&attribute.TestNoItemAttributes{
+				Id: "trsr_one",
+				Attrs: &attribute.TestNoItemAttributes_SubResourceAttributes{
+					SubResourceAttributes: &attribute.TestSubResourceAttributes{
+						Name: "test",
+					},
+				},
+			},
+			&attribute.TestNoItemAttributes{
+				Id: "trsr_one",
+				Attrs: &attribute.TestNoItemAttributes_Attributes{
+					Attributes: func() *structpb.Struct {
+						attrs, _ := structpb.NewStruct(map[string]interface{}{
+							"name": "test",
+						})
+						return attrs
+					}(),
+				},
+			},
+		},
+		{
+			"TestListResourceResponse",
+			nil,
+			&attribute.TestListResourceResponse{
+				Items: []*attribute.TestResource{
+					{
+						Type: "sub_resource",
+						Attrs: &attribute.TestResource_SubResourceAttributes{
+							SubResourceAttributes: &attribute.TestSubResourceAttributes{
+								Name: "test",
+							},
+						},
+					},
+					{
+						Type: "default",
+						Attrs: &attribute.TestResource_Attributes{
+							Attributes: func() *structpb.Struct {
+								attrs, _ := structpb.NewStruct(map[string]interface{}{
+									"name": "test",
+								})
+								return attrs
+							}(),
+						},
+					},
+				},
+			},
+			nil,
+			&attribute.TestListResourceResponse{
+				Items: []*attribute.TestResource{
+					{
+						Type: "sub_resource",
+						Attrs: &attribute.TestResource_Attributes{
+							Attributes: func() *structpb.Struct {
+								attrs, _ := structpb.NewStruct(map[string]interface{}{
+									"name": "test",
+								})
+								return attrs
+							}(),
+						},
+					},
+					{
+						Type: "default",
+						Attrs: &attribute.TestResource_Attributes{
+							Attributes: func() *structpb.Struct {
+								attrs, _ := structpb.NewStruct(map[string]interface{}{
+									"name": "test",
+								})
+								return attrs
+							}(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+				require.Empty(t, cmp.Diff(req, tc.expectHandlerReq, protocmp.Transform()))
+				return tc.handlerResp, nil
+			}
+
+			got, err := subtypes.AttributeTransformerInterceptor(ctx)(ctx, tc.req, nil, handler)
+			require.NoError(t, err)
+			require.Empty(t, cmp.Diff(got, tc.excpetResp, protocmp.Transform()))
+		})
+	}
+}
+
+func TestAttributeTransformerInterceptorRequestErrors(t *testing.T) {
```

Comment:
Could we add a test here as well to see what happens when the response contains an unknown `type`?

Reply:
See d6d2e1f38